### PR TITLE
refactor(os): make hardcode pkgnames dynamic

### LINKED ISSUE #431: salt-minion states fail on FreeBSD
Py2 is hardcoded in osfamilymap
```
          ID: salt-minion
    Function: pkg.installed
        Name: py27-salt
      Result: False
     Comment: The following packages failed to install/update: py27-salt
     Started: 14:59:34.935576
    Duration: 8028.683 ms

Checking integrity... done (6 conflicting)
  - py27-salt-2016.11.2 [SaltStack] conflicts with py36-salt-2019.2.0_2 [installed] on /usr/local/etc/salt/master.sample
  - py27-salt-2019.2.0_2 [FreeBSD] conflicts with py36-salt-2019.2.0_2 [installed] on /usr/local/etc/salt/master.sample
  - py27-tornado-4.4.2 [SaltStack] conflicts with py27-tornado4-4.5.3 [installed] on /usr/local/lib/python2.7/site-packages/tornado/__init__.py
  - py27-setuptools27-32.1.0 [SaltStack] conflicts with py27-setuptools-41.1.0 [installed] on /usr/local/lib/python2.7/site-packages/easy-install.pth.dist
  - py27-setuptools27-32.1.0 [SaltStack] conflicts with py36-setuptools-41.1.0 [installed] on /usr/local/bin/easy_install
  - py27-msgpack-python-0.4.7 [SaltStack] conflicts with py27-msgpack-0.6.1 [installed] on /usr/local/lib/python2.7/site-packages/msgpack/__init__.py
Cannot solve problem using SAT solver, trying another plan
```

Versions
```
Salt Version:
           Salt: 2019.2.0
 
Dependency Versions:
           cffi: 1.12.3
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.1
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.8 (default, Jun 11 2019, 01:16:36)
   python-gnupg: Not Installed
         PyYAML: 5.1
          PyZMQ: 18.1.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1
 
System Versions:
           dist:   
         locale: US-ASCII
        machine: amd64
        release: 11.2-RELEASE
         system: FreeBSD
        version: Not Installed
```

### PR DESCRIPTION
**Pull request by noelmcloughlin**
Resolve #431 and improve os mapping.
Remove "py2: xxx" because it is not referenced by any SLS/Jinja.

### CODE REVIEW

**Comment by noelmcloughlin** [comment]
Hey @daks @aboe76 A favour. Could anyone see anything obvious wrong with this?  Something obvious but I cannot see it...
https://travis-ci.com/saltstack-formulas/salt-formula/jobs/278400151#L1024

**Comment by aboe76** [comment]
@noelmcloughlin I think it's this:
```
{%- if pillar['py_ver'] is defined %}
{%- endif %}
```
it should be:
```
{%- if pillar.get('salt:py_ver') is defined %}
{%- endif %}
```

But why define it twice?
You first set py_ver_repr in line 7 then make an if statement and within the if statement you set a default to the same as the first set...

**Comment by noelmcloughlin** [comment]
Thanks @aboe76   I've got a bit rusty with salt.
> But why define it twice?
Well spotted :-), I was troubleshooting so added a second check, I'll try your suggestion now.

**Comment by daks** [comment]
> @noelmcloughlin I think it's this:
> 
> ```
> {%- if pillar['py_ver'] is defined %}
> {%- endif %}
> ```
> 
> it should be:
> 
> ```
> {%- if pillar.get('salt:py_ver') is defined %}
> {%- endif %}
> ```
> 
> But why define it twice?
> You first set py_ver_repr in line 7 then make an if statement and within the if statement you set a default to the same as the first set...

isn't the format `salt['pillar.get']` the prefered one?

**Comment by aboe76** [comment]
> > @noelmcloughlin I think it's this:
> > ```
> > {%- if pillar['py_ver'] is defined %}
> > {%- endif %}
> > ```
> > 
> > 
> > it should be:
> > ```
> > {%- if pillar.get('salt:py_ver') is defined %}
> > {%- endif %}
> > ```
> > 
> > 
> > But why define it twice?
> > You first set py_ver_repr in line 7 then make an if statement and within the if statement you set a default to the same as the first set...
> 
> isn't the format `salt['pillar.get']` the prefered one?

@noelmcloughlin yes that's the prefered one!

File: test/salt/pillar/v201707-py2.sls [hunk 1]
```diff
@@ -3,4 +3,4 @@
 ---
 salt:
   release: '2017.7'
-  py_ver: 'py2'
+  py_ver: 'apt'
```
Comment (aboe76) [inline test/salt/pillar/v201707-py2.sls]: this shouldn't be 'apt'

File: test/salt/pillar/v201803-py2.sls [hunk 2]
```diff
@@ -3,4 +3,4 @@
 ---
 salt:
   release: '2018.3'
-  py_ver: 'py2'
+  py_ver: 'apt'
```
Comment (aboe76) [inline test/salt/pillar/v201803-py2.sls]: this shouldn't be 'apt'

File: test/salt/pillar/v201902-py2.sls [hunk 3]
```diff
@@ -3,4 +3,4 @@
 ---
 salt:
   release: '2019.2'
-  py_ver: 'py2'
+  py_ver: 'apt'
```
Comment (aboe76) [inline test/salt/pillar/v201902-py2.sls]: this shouldn't be 'apt'

**Comment by noelmcloughlin** [comment]
Closing this to rebase.

**Pull request closed without merging**